app/src/main/java/com/example/geoquiz/QuizActivity.kt

```diff
@@ -0,0 +1,196 @@
+
+
+        }
+
+        prevButton.setOnClickListener {
+            quizViewModel.moveToPrev()
+            updateQuestion()
+
+        }
+
+        updateQuestion()
+
+        scoreButton.setOnClickListener {
+
+            for (i in correctAnswers) {
+                score += 1
+            }
+
+            val scoreFinalValue=(score.toDouble()/quizViewModel.questionBank.size)*100
+
+
+            Toast.makeText(
+                this,
+                "Your score is: ${scoreFinalValue}%",
+                Toast.LENGTH_SHORT
+            ).show()
+
+        }
+
+        cheatButton.setOnClickListener {
+            val answerIsTrue = quizViewModel.currentQuestionAnswer
+            val intent = CheatActivity.newIntent(this@QuizActivity, answerIsTrue)
+            startActivityForResult(intent, REQUEST_CODE_CHEAT)
+        }
+
+
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (resultCode != Activity.RESULT_OK) {
+            return
+        }
+        if (requestCode == REQUEST_CODE_CHEAT) {
+            quizViewModel.isCheater = data?.getBooleanExtra(EXTRA_ANSWER_SHOWN, false) ?: false
+        }
+    }
+
+    override fun onStart() {
+        super.onStart()
+        Log.d(TAG, "onStart() called")
+    }
+
+    override fun onResume() {
+        super.onResume()
+        Log.d(TAG, "onResume() called")
+
+    }
+
+    override fun onPause() {
+        super.onPause()
+        Log.d(TAG, "onPause() called")
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        Log.d(TAG, "onSaveInstanceState() called")
+        outState.putInt(QUESTION_INDEX_KEY, quizViewModel.currentIndex)
+        outState.putBooleanArray(QUESTIONS_ANSWERED_KEY, questionWasAnswered)
+    }
+
+    override fun onStop() {
+        super.onStop()
+        Log.d(TAG, "onStop() called")
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        Log.d(TAG, "onDestroy() called")
+    }
+
+
+    private fun updateQuestion() {
+        val questionTextResId = quizViewModel.currentQuestionText
+        questionTextView.setText(questionTextResId)
+        falseButton.isClickable = !(questionWasAnswered[quizViewModel.currentIndex]!!)
+        trueButton.isClickable = !(questionWasAnswered[quizViewModel.currentIndex]!!)
+
+    }
+
+    private fun checkAnswer(userAnswer: Boolean) {
+        correctAnswers = ArrayList()
+        val answerIsTrue = quizViewModel.currentQuestionAnswer
+
+
+        val messageResId = when {
+            quizViewModel.isCheater -> R.string.judgment_toast
+            userAnswer == answerIsTrue -> R.string.correct_toast
+
+            else -> R.string.incorrect_toast
+        }
+
+
+        Toast.makeText(this, messageResId, Toast.LENGTH_SHORT).show()
+        questionWasAnswered[quizViewModel.currentIndex] = true
+        trueButton.isClickable = false
```

Comment:
I would couple these isClickable into a method named `setAnswerButtonsClickable` it's much easier to understand and this is used in two places